crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::code;
+
+use crate::{error::graphql_error, types::transaction_exec::ExecutionResult};
+use async_graphql::*;
+use fastcrypto::encoding::Encoding;
+use fastcrypto::{encoding::Base64, traits::ToFromBytes};
+use shared_crypto::intent::Intent;
+use sui_json_rpc_types::SuiTransactionBlockResponseOptions;
+use sui_sdk::SuiClient;
+use sui_types::quorum_driver_types::ExecuteTransactionRequestType;
+use sui_types::{signature::GenericSignature, transaction::Transaction};
+
+pub struct Mutation;
+
+#[Object]
+impl Mutation {
+    /// Execute a transaction, committing its effects on chain.
+    ///
+    /// `txBytes` is a `TransactionData` struct that has been BCS-encoded
+    ///     and then Base64-encoded.
+    /// `signatures` are a list of `flag || signature || pubkey` bytes,
+    ///     Base64-encoded.
+    ///
+    /// Waits until the transaction has been finalised on chain to return
+    /// its transaction digest.  If the transaction could not be
+    /// finalised, returns the errors that prevented it, instead.
+    async fn execute_transaction_block(
+        &self,
+        ctx: &Context<'_>,
+        tx_bytes: String,
+        signatures: Vec<String>,
+    ) -> Result<ExecutionResult> {
+        let sui_sdk_client: &SuiClient = ctx.data().map_err(|_| {
+            graphql_error(
+                code::INTERNAL_SERVER_ERROR,
+                "Unable to fetch Sui SDK client",
+            )
+        })?;
+
+        let tx_data = bcs::from_bytes(&Base64::decode(&tx_bytes).map_err(|e| {
+            graphql_error(
+                code::INTERNAL_SERVER_ERROR,
+                format!("Unable to deserialize transaction bytes from Base64: {e}"),
+            )
+        })?)?;
```

Comment:
I forgot on the previous review: we shouldn't be using the `graphql_error` construct outside of extensions, instead we need to use the crate's `Error` type and the `extend` function, so that the errors get extended with both a code and a position.  In this case, these aren't internal errors -- it's a user input error if the Base64 or BCS is bad:

```suggestion
        let sui_sdk_client: &SuiClient = ctx.data()
            .map_err(|_| Error::Internal("Unable to fetch Sui SDK client".to_string()))
            .extend()?

        let tx_data = bcs::from_bytes(
            &Base64::decode(&tx_bytes)
                .map_err(|e| Error::Client(format!("Unable to deserialize transaction bytes from Base64: {e}")))
                .extend()?
        )
        .map_err(|e| Error::Client(format!("Unable to deserialize transaction bytes as BCS: {e}"))
        .extend()?;
```

In this case, it would probably be helpful to create something like:

Reply:
Similar comments below -- very few of these are actually internal errors.  Most of them are client errors, and errors due to execution might need a new code altogether.  I couldn't find a standard GraphQL code that applies, but I think it's fine to invent a new one for this case, because execution is quite specific to our use case.